# Reading the dataset to python environment 

In [ ]:
import pandas as pd
data=pd.read_csv('Advertising.csv')

# 1. Explain the dataset

In [5]:
data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


As per above the data is having an unnamed column which is actually serial no of each row and need to be dropped. There are a total of 200 samples. The data is about advertising done through different media and the sales achieved.

In [7]:
data.isna().sum()

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

There is no invalid data in the columns and no null values too



# 2. Check the summary statistics and discuss the max, min, avg, median, and percentiles

In [9]:
data.describe(exclude=[int])

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

# a)Minimum values of data

In [19]:
import numpy as np
minimum = np.min(data)
print(minimum)

Unnamed: 0    1.0
TV            0.7
Radio         0.0
Newspaper     0.3
Sales         1.6
dtype: float64


In the Tv column ,minimum value is 0.70 ; for Radio column,minimum value is 0; for Newspaper column,minimum value is 0.30 and for sales column minimum value is 1.60 Comparing minimum advertisement expenditure across columns,Radio comes first(lowest spending), News paper comes second and TV comes third.



# b)Maximum values of data

In [12]:
maximum = np.max(data)
print(maximum)

Unnamed: 0    200.0
TV            296.4
Radio          49.6
Newspaper     114.0
Sales          27.0
dtype: float64


Advertisement in TV has maximum spending in one instance

# c)Average values of data

In [13]:
data.mean()

Unnamed: 0    100.5000
TV            147.0425
Radio          23.2640
Newspaper      30.5540
Sales          14.0225
dtype: float64

Average value in the TV column is ≈147.04,average value in the Radio column is ≈23.26,average value in the Newspaper column is ≈30.55 and average value in the sales column is ≈14.02 Comparing mean advertisement expenditure across columns,Tv occupies first position, Newspaper occupies second position and Radio third position.

# d)Median values of data

In [14]:
data.median()

Unnamed: 0    100.50
TV            149.75
Radio          22.90
Newspaper      25.75
Sales          12.90
dtype: float64

Advertisement in TV has larger spendings compared to other media

# e)Percentiles values of data

In [20]:
data.quantile([.25,.5,.75])

Unnamed: 0       TV   Radio  Newspaper   Sales
0.25       50.75   74.375   9.975      12.75  10.375
0.50      100.50  149.750  22.900      25.75  12.900
0.75      150.25  218.825  36.525      45.10  17.400

Data shows that the spread of advertisement on TV is much wider than other medias

# 3.The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings

Manager wants to find whether same amount was spent for three advertisements. For that we need to check whether amount spent of all modes of adverisements are same or not using oneway Anova

In [ ]:
# Hypothesis:

H0: Amount Spent on three Advertisements are same.,µ1=µ2=µ3
H1: Amount spent on three advertisements are different,All μ are not equal

In [ ]:
#setting the significance level

Considering significance level(α)=0.05 or 5%


In [ ]:
# Determining p value using one way anova (for comparing means from 3 groups)

In [26]:
import numpy as np
from scipy.stats import f_oneway
x=np.array(data['TV'])
y=np.array(data['Radio'])
z=np.array(data['Newspaper'])
f_oneway(x,y,z)

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

In [ ]:
# comparing p value with significance level(0.05) 

p value: 4.552931539744962e-103 ≈ 4.55 x 10^-103 < 0.05

Rejecting the null hypothesis

In [ ]:
#Conclusion

Since p value is less than the selected significance level of 0.05 ,rejecting the null hypothesis which means that the amount spent for atleast one advertisement differs from the rest i.e, All μ are not equal